### PR TITLE
DAOS-12853 common: More fixes for storage estimator

### PR DESCRIPTION
**Pull request by jolivier23**
Fix documentation link
Remove array metadata from dfs file entry, DFS doesn't create an entry in the file object.
Make dfs entry size reflect actual size
Don't sort yaml keys
Fix output to use KiB, MiB, etc

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by mchaarawi — APPROVED** [review]

File: src/vos/storage_estimator/common/vos_size.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 def convert(stat):
     """Convert byte value to pretty string"""
     size = 1024 * 1024 * 1024 * 1024 * 1024
-    for mag in ['P', 'T', 'G', 'M', 'K']:
+    for mag in ['PiB', 'TiB', 'GiB', 'MiB', 'KiB', '  B']:
```
Comment (daltonbohning) [inline src/vos/storage_estimator/common/vos_size.py:13]: Is `' B'` intentional? I.e. should there be an extra space?

Comment (jolivier23) [reply src/vos/storage_estimator/common/vos_size.py:13]: yeah, just looks better in the output if they all take 3 spaces.

Comment (jolivier23) [reply src/vos/storage_estimator/common/vos_size.py:13]:
```
Using DAOS version: 2.3.103
  Reading VOS structures from current installation
Metadata breakdown:
	pool                :     320.00   B ( 0.01%)
	container           :    1024.00   B ( 0.02%)
	object              :     196.75 KiB ( 3.73%)
	dkey                :     308.01 KiB ( 5.84%)
	akey                :     330.67 KiB ( 6.26%)
	single_value        :       1.12 KiB ( 0.02%)
	array               :     192.19 KiB ( 3.64%)
	user_meta           :      32.03 KiB ( 0.61%)
	total_meta          :       1.01 MiB (19.51%)
Data breakdown:
	total_meta          :       1.01 MiB (19.51%)
	user_value          :       4.12 MiB (79.88%)
	total               :       5.15 MiB (100.00%)
Physical storage estimation:
	scm_total           :       1.15 MiB (22.40%)
	nvme_total          :       4.00 MiB (77.60%)
Total storage required:       5.15 MiB
```

Comment (jolivier23) [reply src/vos/storage_estimator/common/vos_size.py:13]: I see there is a bug though...Should be stat >= size

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**